src/main/java/org/opensrp/repository/postgres/mapper/custom/xml/CustomLocationMetadataMapper.xml

```diff
@@ -141,4 +141,40 @@
 		LIMIT #{limit} OFFSET #{offset}
 	</select>
 
+	<select id="selectWithChildren"
+			resultMap="org.opensrp.repository.postgres.mapper.LocationMapper.BaseResultMap">
+
+		select
+		<if test="example.distinct">
+			distinct
+		</if>
+
+		<if test="geometry">
+			<include refid="Base_Column_List" />
+		</if>
+
+		<if test="!geometry">
+			<include refid="No_Geometry_Column_List" />
+		</if>
+		from core.location l
+		Where l.id in
+		(
+		WITH RECURSIVE locations
+		AS
+		( SELECT lm1.geojson_id, lm1.parent_id, lm1.location_id
+		FROM core.location_metadata lm1 WHERE geojson_id = #{locationId,jdbcType=VARCHAR}
+		UNION
+		SELECT lm2.geojson_id, lm2.parent_id, lm2.location_id
+		FROM core.location_metadata lm2, locations lm1
```

Comment:
Why not use inner join more explicitly instead of joining in the where clause e.g
     ```
 FROM core.location_metadata lm2 join locations lm1
                                      on lm2.parent_id = lm1.geojson_id
```